Source/Extensions/Blazorise.SignaturePad/SignaturePad.razor.cs

```diff
@@ -0,0 +1,283 @@
+    }
+
+    /// <inheritdoc/>
+    protected override async ValueTask DisposeAsync( bool disposing )
+    {
+        if ( disposing && Rendered )
+        {
+            await JSModule.SafeDestroy( ElementRef, ElementId );
+
+            await JSModule.SafeDisposeAsync();
+
+            if ( DotNetObjectRef != null )
+            {
+                DotNetObjectRef.Dispose();
+                DotNetObjectRef = null;
+            }
+        }
+
+        await base.DisposeAsync( disposing );
+    }
+
+    /// <summary>
+    /// Clears the content of a signature canvas.
+    /// </summary>
+    /// <returns>A task that represents the asynchronous operation.</returns>
+    public async Task Clear()
+    {
+        if ( JSModule != null )
+        {
+            await JSModule.Clear( ElementRef, ElementId );
+        }
+    }
+
+    /// <summary>
+    /// This method is called by JavaScript when a stroke has ended in the signature pad. It takes the encoded image
+    /// from the signature pad, converts it to bytes and sets the Value property of the component to the image data.
+    /// It then invokes the ValueChanged and EndStroke events asynchronously to notify any subscribers of the change.
+    /// </summary>
+    [JSInvokable]
+    public async Task NotifyEndStroke( string value )
+    {
+        if ( string.IsNullOrEmpty( value ) )
+            return;
+
+        var valueParts = value.Split( ',' );
+
+        if ( valueParts.Length > 1 )
+        {
+            Value = Convert.FromBase64String( valueParts[1] );
+
+            await ValueChanged.InvokeAsync( Value );
+            await EndStroke.InvokeAsync( Value );
+        }
+    }
+
+    /// <summary>
+    /// This method is called by JavaScript when a new stroke has begun in the signature pad. It invokes the BeginStroke
+    /// event asynchronously to notify any subscribers of the event.
+    /// </summary>
+    [JSInvokable]
+    public async Task NotifyBeginStroke()
+    {
+        await BeginStroke.InvokeAsync();
+    }
+
+    /// <summary>
+    /// Gets the string representation of the <see cref="ImageType"/>.
+    /// </summary>
+    /// <param name="imageType">Image type value.</param>
+    /// <returns>String representation of the <see cref="ImageType"/>.</returns>
+    private static string ToImageTypeString( SignaturePadImageType imageType ) => imageType switch
+    {
+        SignaturePadImageType.Jpeg => "jpeg",
+        SignaturePadImageType.Svg => "svg",
+        SignaturePadImageType.Png => "png",
+        _ => "png",
+    };
+
+    #endregion
+
+    #region Properties
+
+    /// <summary>
+    /// Reference to the object that should be accessed through JSInterop.
+    /// </summary>
+    protected DotNetObjectReference<SignaturePad> DotNetObjectRef { get; private set; }
+
+    /// <summary>
+    /// Gets or sets the <see cref="JSSignaturePadModule"/> instance.
+    /// </summary>
+    protected JSSignaturePadModule JSModule { get; private set; }
```

Comment:
Same answer 

> Most, if not all, of extensions, are done like this. If would to add a DI support, it would mean that we also have to have a register method at Startup.cs, eg. .AddSignaturePad(). Unless we really need it, I tend to avoid it.

Reply:
I see. Nice approach then.